pkg/pillar/cmd/downloader/syncop.go

```diff
@@ -115,8 +114,10 @@ func handleSyncOp(ctx *downloaderContext, key string,
 		}
 		trType = zedUpload.SyncAwsTr
 		serverURL = dsCtx.DownloadURL
-		remoteName = filename
-		metricsUrl = fmt.Sprintf("S3:%s/%s", dsCtx.Dpath, filename)
+		// pass in the config.Name instead of 'filename' which
+		// does not contain the prefix of the relative path with '/'s
+		remoteName = config.Name
+		metricsUrl = fmt.Sprintf("S3:%s/%s", dsCtx.Dpath, config.Name)
```

Comment:
Why do we change the metricsUrl here but in none of the other cases?

Reply:
For other cases, ```metricsUrl``` is the download URL but for S3 we are overwriting it with a custom string.

Reply:
OK